src/components/cards/Cards.js

```diff
@@ -1,20 +1,48 @@
+import { useEffect, useState } from 'react'
 import Card from './card/Card'
 import './Cards.css'
-import { getDataArr } from './getDataArr'
 
 const Cards = props => {
-  let dataArr = getDataArr()
-  console.log('Cards-dataArr is of type: ' + typeof dataArr)
+  const [dataArr, setDataArr] = useState(null)
+  const [loading, setLoading] = useState(true)
+  const [error, setError] = useState(null)
 
-  if (props.house !== 'all') {
-    dataArr = dataArr.filter(
-      person => person.house.toLowerCase() === props.house
-    )
+  useEffect(() => {
+    fetch('http://hp-api.herokuapp.com/api/characters')
+      .then(response => {
+        if (response.ok) {
+          return response.json()
+        }
+        throw response
+      })
```

Comment:
I don't think the `throw response` is necessary. 🤔 If it is not okay, then it will be caught by the `.catch` I think.
```suggestion
      .then(response => response.json())
```

Reply:
> Within the first .then() callback, we check to see if the response was okay (response.ok). If so, we return our response to pass to the next, then call back as JSON data, since that's the data we'll get back from our random user API.
> 
> If it's not an okay response, we assume there was an error making the request. Using fetch, we need to handle the errors ourselves, so we throw response as an error for it to handled by our catch callback.
> [Source](https://www.freecodecamp.org/news/fetch-data-react/)

Tbh, I dont know. I found it on freecodecamp like this and I thought that the reason why they are using it makes sense.

Reply:
Hm, yea, I am not an expert about best practices here as well. 😅 Just the way I learned it is the suggestion I have made, but as long as it is working, it should be okay.